### ISSUE TITLE
Reverse foreign key lookups

### ISSUE THREAD

**Issue by dantownsend** [piccolo-orm/piccolo#378]
Now we have M2M, it makes sense to add something similar for reverse foreign key lookups.

This needs doing before v1.

Here's an example API design:

```python

class Manager(Table):
    name = Varchar()
    bands = ReverseLookup(LazyTableReference('Band', module_path=__module__))


class Band(Table):
    name = Varchar()
    manager = ForeignKey(Manager)


>>> await Manager.select(Manager.name, Manager.bands(Band.name, as_list=True))
[
    {
        "name": "Guido",
        "bands": ["Pythonistas"]
    }
]
```

**Comment by AliSayyah** [comment]
@dantownsend Is it necessary to have a ReverseLookup field specified by the user?

**Comment by dantownsend** [comment]
@AliSayyah Good point - it could be done that way. The reverse lookup could then be created by the table metaclass. The only downside is it might not work as well with tab completion, but it's a simpler API for users.

**Comment by brnosouza** [comment]
And what about stubs for that?

**Comment by dantownsend** [comment]
One option is for the user to do something like this, which would give better tab completion:

```python
class Manager(Table):
   name = Varchar()
   bands: ReverseLookup


class Band(Table):
    name = Varchar()
    manager = ForeignKey(Manager, reverse_lookup='bands')
```

**Comment by sinisaos** [comment]
@dantownsend We already have all the tools for reverse lookup without intervention in the source code. If we document properly, users will only have to write a few lines of code. My examples will be `async` FastAPI methods, but everything works as `sync` too. Here is a [gist](https://gist.github.com/sinisaos/5d9e7b36d42d6e125d93177a370dc7f4) with all 4 possibilities. What do you think about that?

**Comment by dantownsend** [comment]
@sinisaos I looked at the gist - what you've done is similar to how I currently do it.

I think having something built in would be nice. We'll get around to it eventually!

You could put a page here if you like: https://piccolo-orm.readthedocs.io/en/latest/piccolo/tutorials/index.html

**Comment by sinisaos** [comment]
@dantownsend Thank you for your reply. No need to add this to the docs because if anyone needs a workaround, they can find it here. I've updated the gist with your suggestion so it's all there.
